core/src/banking_stage.rs

```diff
@@ -1203,6 +1292,68 @@ impl BankingStage {
             .collect()
     }
 
+    fn generate_vote_packet_indexes(vers: &PinnedVec<Packet>) -> Vec<usize> {
+        vers.iter()
+            .enumerate()
+            .filter_map(|(index, ver)| {
+                if !ver.meta.discard {
+                    let tx: Transaction = limited_deserialize(&ver.data[0..ver.meta.size]).ok()?;
+                    if is_simple_vote_transaction(&tx) {
```

Comment:
I think full deserialization is probably overkill here. I think at a minimum we want to know that the tx has:
1. only 1 instruction
2. the instruction calls the vote program

How about we do the above checks in sigverify and then tag packets like we do for `is_tracer_tx`?

Reply:
It's probably fine (and good) to prioritize non "Vote" vote program instructions through like "Init", "Authorize", etc as well

Reply:
+1 probably fine to prioritize other Vote instructions.
Painful to find the instructions and then the offsets manually. Basically wrote our own custom Transaction deserializer at that point. But it probably would be a lot faster especially with a large Transaction with a bunch of Vecs to allocate and copy.

Reply:
Yeah it's painful, zero copy deserialization would be really nice for this cc @t-nelson

Reply:
I bet we can `bytemuck` here for a partial zero copy deserialization of the parts of the transaction we need to inspect

Reply:
I think we need the fields to be byte aligned for that

Reply:
🤔 ya, dang maybe not then

Reply:
yeah, manually count offset isn't fun, but doable. Doing this right now, hope to make bench-tps better.

Reply:
made suggested change, improved `bench-tps`, it is now back to v1.6 level. Please review the change